applications/ConvectionDiffusionApplication/custom_utilities/bfecc_convection.h

```diff
@@ -68,9 +68,15 @@ template<std::size_t TDim> class BFECCConvection
         std::vector< Vector > Ns( rModelPart.Nodes().size());
         std::vector< bool > found( rModelPart.Nodes().size());
 
-        // Allocate non-historical variables
-        for (auto &r_node : rModelPart.Nodes()) {
-            r_node.SetValue(rVar, 0.0);
+        // Allocate non-historical variables and update old velocity as per dt_factor
+        #pragma omp parallel for
+        for (int i = 0; i < nparticles; i++) {
+            auto iparticle = rModelPart.NodesBegin() + i;
+            iparticle->SetValue(rVar, 0.0);
```

Comment:
This is not needed as you are overwriting the zero value below.

Reply:
The non-historical variable do not need to be initialized? Maybe, if I use SetValue instead of GetValue below?

Reply:
More than initialized, they need to be allocated. Each time you do a `SetValue` you allocate. As you need in here the `SetValue` below, since you cannot ensure that the variable is already allocated, the `SetValue` to zero can be removed. 

This is one reason to make the BFECC convection util more process-like.

Reply:
So, I think it is better to keep this SetValue here and move it to the initialization in its future clean process-like implementation. Let's mark it as a TODO for future.